_posts/2017-06-09-Principles-of-object-oriented-design.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+author: domenique
+disqus: false
+header-img: "img/patterns.jpg"
+---
+
+_SOLID_, The famous acronym coined by Micheal Feathers describing the first five principles of object oriented design composed by Robert C. Martin a.k.a Uncle Bob. What most people tend to forget is that __SOLID describes only five of the in total eleven principles of object oriented design__. These principles date all the way back to 1995, yes Uncle Bob is really that old :)
+
+This post will briefly describe the well known SOLID principles, after which we will dive deeper into the 6 other, lesser known principles. This post will ultimately become a series, so do check back for follow-up articles.
+
+# SOLID Principles
+
+The SOLID principles are the first five principles of object oriented design and focus around class design.
+
+- **S**ingle responsibility principle
+- **O**pen closed principle
+- **L**iskov substitution principle
+- **I**nterface segregation principle
+- **D**ependency inversion principle
+
+# Single responsibility principle
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/2k1uOqRb0HU" frameborder="0" allowfullscreen="">
+</iframe>
+
+__SRP__(Single Responsibility Principle) dictates that a class should have only **one responsibility and this responsibility should be fully encapsulated by the class**.
+
+In the context of __SRP__, Uncle bob refers to a responsibility as being _a reason for change_. He concluded that if you can find more then one __motive for changing__ a class, then that class has more then one responsibility. Failing to adhere to this principle could result in _rigidity_ because clients of your class will be __forced to change for reasons they shouldn't__, on the other hand, applying this principle without looking carefully at the context of your application can have a negative effect and introduce needless complexity.
+
+> The Single Responsibility Principle is one of the simplest of the principles, and one of the hardest to get right. Con-joining responsibilities is something that we do naturally. Finding and separating those responsibilities from one another is much of what software design is really about. Indeed, the rest of the principles we will discuss come back to this issue in one way or another.
+>
+> -- Uncle Bob, Agile Software Development. Principles, patterns and practices
```

Comment:
SRP is ook iets dat je op meerdere niveaus kunt hebben. Op een domain, een use case, een aggregate een functie, een lijn of code. Gerelateerd aan niveaus van abstracties.